omniscidb/QueryEngine/Execute.cpp

```diff
@@ -1831,8 +1835,25 @@ hdk::ResultSetTable Executor::executeWorkUnitImpl(
     exe_policy = std::make_unique<policy::FragmentIDAssignmentExecutionPolicy>(
         ExecutorDeviceType::CPU);
   } else {
+#ifdef HAVE_COST_MODEL
     costmodel::DummyCostModel cost_model(device_type, config_->exec.heterogeneous);
```

Comment:
The idea behind having a dummy cost model was to not pollute the code with macros. Can we have the model library handle it internally and return a round-robin policy (or whichever else requested)?

Reply:
that's exactly what we're doing in `predict`. define is here to handle the case when we don't use a library with a cost model. alternatively, we can assume that we always use it and remove the ability to turn it off in cmake.
or maybe I misunderstood the comment?